### PR TITLE
fix: `isConstruct` is wrong when symlinking libraries

### PR DESCRIPTION
**Pull request by rix0rrr**
`Construct.isConstruct` was still using (and recommending) `instanceof`,
even though `instanceof` can never be made to work reliably.

When we thought `instanceof` was safe to use again, it's because we
thought that `npm install` combined with `peerDependencies` would
make sure only one copy of `constructs` would ever be installed.

That's correct, but monorepos and users using `npm link` can
still mess up that guarantee, so we cannot rely on it after all.

### CODE REVIEW

File: src/construct.ts
Comment (Chriscbr) [inline src/construct.ts:560]: Why not put this in the constructor? (is there any difference?)

**Review by Chriscbr — APPROVED** [review]
I've been pondering for a bit if there's any ways this could cause a regression, but I can't think of anything...

Approving with do-not-merge so you can look one comment I added. Since this library is used in a lot of places, though, it might not hurt to get a second pair of eyes on this. :)

File: src/construct.ts
Comment (rix0rrr) [reply src/construct.ts:560]: The difference is where the symbol lives: on the *instance* or on the *class*.

This puts it on the class, putting it in the construct (and presumably passing `this` instead of `Construct.prototype`) would put it on the instance.

Putting it on the class saves an instruction on every construct instantiation, and the memory of saving it in every symbol table (plus it might potentially have some impact on V8s optimizations... though I'm for sure not looking at any of that!)

It's probably a minor thing, but this ought to be slightly more efficient.

**Review by MrArnoldPalmer — APPROVED** [review]
Makes sense to me as far as "another pair of eyes" 👀

**Pull request merged**
